tests/unit/name-request-mixin.spec.ts

```diff
@@ -143,6 +146,43 @@ describe('Name Request Mixin', () => {
 
     nr.request_action_cd = 'CNV'
     expect(vm.isNrValid(nr)).toBe(true)
+
+    nr.request_action_cd = 'NEW'
+    expect(vm.isNrValid(nr)).toBe(false)
+  })
+
+  it('identifies valid and invalid NRs for firms', () => {
+    store.state.stateModel.tombstone.entityType = 'SP'
+
+    let nr = null
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr = {}
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr.state = 'APPROVED'
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr.expirationDate = '2021-11-05T07:01:00+00:00'
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr.names = [{ state: 'APPROVED', name: 'name' }]
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr.nrNum = 'NR 1234567'
+    expect(vm.isNrValid(nr)).toBe(false)
+
+    nr.requestTypeCd = 'CR'
```

Comment:
Wrong request type code for a firm NR.

Reply:
Sorry, I see you're testing the "false" case here.

However, I don't see where you set a firm request type code in here, so you're not actually testing firm NR validation.